WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -126,6 +126,14 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 
 			uint256 txId = tx.GetHash();
 
+			// If we already have the transaction, then let's work on that.
+			if (TransactionStore.TryGetTransaction(txId, out var foundTx))
+			{
+				foundTx.TryUpdate(tx);
+				tx = foundTx;
+				result = new ProcessedResult(tx);
+			}
+
```

Comment:
This might come with some performance hit at wallet re-scan. During normal wallet load there's no performance issue.

Reply:
Rescan takes a lot of time anyway and it should not happen in normal circumstances. So this is fine.